sim/common/wotlk/enchant_effects.go

```diff
@@ -197,6 +195,22 @@ func init() {
 				}
 			},
 		})
+
+		core.RegisterOnItemSwap(3789, func(sim *core.Simulation) {
+			mh = character.Equip[proto.ItemSlot_ItemSlotMainHand].Enchant.EffectID == 3789
+			oh = character.Equip[proto.ItemSlot_ItemSlotOffHand].Enchant.EffectID == 3789
+
+			procMask = core.GetMeleeProcMaskForHands(mh, oh)
+			ppmm = character.AutoAttacks.NewPPMManager(1.0, procMask)
+			if ppmm.Chance(procMask) == 0 {
+				aura.Deactivate(sim)
+				return
```

Comment:
This `return` could be removed and replaced with an `else` clause. Same comment for many of the other effects.

Guard clauses (https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) are better for readability when there are several different cases, but when there are only 2 its usually easier to understand with an if-else.